More travis

joshmoore opened the pull request:
Builds on gh-2 by moving from a "release" (i.e. all roles on galaxy) to a "latest" (i.e. all roles on github) strategy. Note: this is in preparation either for more thorough scanning tools (like https://github.com/nickjj/ansigenome#example-rebuild-output) or our own jenkins job for a "merge" strategy.

joshmoore committed:
Add generation code

joshmoore committed:
Update with all repos on github

joshmoore committed:
Simplify update.py

joshmoore commented:
Current state needs special omero-common handling and then:

 * https://galaxy.ansible.com/openmicroscopy/cadvisor/
 * https://galaxy.ansible.com/openmicroscopy/mysql-backup/
 * https://galaxy.ansible.com/openmicroscopy/system-monitor-agent/

joshmoore committed:
Add workaround for galaxy-issues/issues#259

joshmoore committed:
Remove omero-common completely

manics commented on .travis.yml:
```diff
@@ -19,54 +19,53 @@ script:
 - mv openmicroscopy.$ROLE ansible-role-$ROLE
 - cd ansible-role-$ROLE
-# TODO: Some of these roles aren't using molecule
-- if [[ -f molecule.yml ]]; then molecule test; fi
```
I guess not running `molecule test` is one way to ensure travis passes 😃

joshmoore replied:
Doh.

joshmoore replied:
Should have known it was too good to be true.

joshmoore committed:
Re-activate molecule!

joshmoore commented:
An aside while I work on getting this PR green since it may take a while:

```
ome-ansible-molecule-dependencies jamoore$ mkdir vendor; time for x in $(./repos.sh); do git clone --depth=1 git://github.com/openmicroscopy/$x vendor/openmicroscopy.${x##ansible-role-}; done
...
real	0m33.391s
user	0m0.412s
sys	0m1.119s
```
might be a fairly nice way to "get all 

cc: @openmicroscopy/devops

joshmoore committed:
Don't require molecule tests

Now for all non-molecule roles, we fallback to the previous
simple syntax check. This is so that this PR can become green
so we can test a bump to Ansible 2.3.1.

joshmoore committed:
Remove 2 roles with missing test infra

joshmoore committed:
Refresh passing projects

joshmoore committed:
Oomit nfs-share: molecule is vagrant only

manics commented on update.py:
```diff
@@ -0,0 +1,23 @@
+        role = role[13:]
+    replacement += "- ROLE=%s\n" % role
+replacement += "%s" % TAG
+print re.sub("%s.*?%s" % (TAG, TAG), txt, replacement, flags=re.DOTALL),
```
This script never makes any changes. If I delete some some `- ROLE=...` lines from `.travis.yml`
`diff <(./repos.sh | ./update.py ) .travis.yml` returns no difference.

joshmoore replied:
My simplification apparently is broken. I've rolled it back.

joshmoore committed:
Revert "Simplify update.py"

This reverts commit 350a9f5131cc0018e1bcd1e0ef79e7292f0ea2c7.

joshmoore commented:
Relaunched travis after a previously unseen issue:

```
=================================== FAILURES ===================================
_____________ test_omero_managedrepo[docker://omero-server-newdep] _____________
File = <class 'testinfra.modules.base.GNUFile'>
    def test_omero_managedrepo(File):
        d = File('/OMERO/ManagedRepository')
        assert d.is_directory
        assert d.user == 'omero-server'
        assert d.group == 'importer'
>       assert d.mode == 0o2775
tests/test_default.py:75:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <file /OMERO/ManagedRepository>
    @property
    def mode(self):
        # Supply a base of 8 when parsing an octal integer
        # e.g. int('644', 8) -> 420
>       return int(self.check_output("stat -c %%a %s", self.path), 8)
E       ValueError: invalid literal for int() with base 8: ''
../../../../virtualenv/python2.7.12/lib/python2.7/site-packages/testinfra/modules/file.py:209: ValueError
```

manics commented:
```
$ diff <(./repos.sh | ./update.py ) .travis.yml
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  594k  100  594k    0     0   497k      0  0:00:01  0:00:01 --:--:--  497k
26d25
< - ROLE=analysis-tools
30,32d28
< - ROLE=cli-utils
< - ROLE=debug-dumpallvars
< - ROLE=devspace
35d30
< - ROLE=haproxy
44d38
< - ROLE=munin-node
48d41
< - ROLE=nfs-share
53d45
< - ROLE=omero-common
59d50
< - ROLE=omero-web-apps
63,65d53
< - ROLE=prometheus
< - ROLE=prometheus-jmx
< - ROLE=prometheus-node
```

I think the following should work with the current dependencies, are they omitted for a reason?
```
- ROLE=omero-common
- ROLE=omero-web-apps
- ROLE=prometheus-jmx
- ROLE=prometheus-node
```

joshmoore commented:
Prometheus only because they are newer than this PR. web-apps I fixed during this repo. common I'm no longer sure about.

joshmoore commented:
Will handle other repos via follows them.
